lib/manageiq/password.rb

```diff
@@ -23,58 +22,46 @@ def initialize(str = nil)
       @encStr = encrypt(str)
     end
 
-    def encrypt(str, ver = "v2", key = self.class.keys[ver])
-      value = key.encrypt64(str).delete("\n") unless str.nil? || str.empty?
-      "#{ver}:{#{value}}"
-    end
+    def self.encrypt(str, key = self.key)
+      return str if str.nil?
 
-    def decrypt(str, legacy = false)
-      if str.nil? || str.empty?
-        str
-      else
-        ver, enc = self.class.split(str)
-        return "" if enc.empty?
+      enc = key.encrypt64(str).delete("\n") unless str.empty?
+      wrap(enc)
+    end
 
-        ver ||= "0" # if we don't know what it is, just assume legacy
-        key_name = (ver == "2" && legacy) ? "alt" : "v#{ver}"
+    def self.decrypt(str, key = self.key)
+      enc = unwrap(str)
+      return enc if enc.nil? || enc.empty?
 
-        begin
-          self.class.keys[key_name].decrypt64(enc).force_encoding('UTF-8')
-        rescue
-          raise PasswordError, "can not decrypt v#{ver}_key encrypted string"
-        end
+      begin
+        key.decrypt64(enc).force_encoding('UTF-8')
+      rescue
+        raise PasswordError, "cannot decrypt encrypted string"
       end
     end
 
-    def recrypt(str)
-      return str if str.nil? || str.empty?
-      decrypted_str =
-        begin
-          # if a legacy v2 key exists, give decrypt the option to use that
-          decrypt(str, self.class.keys["alt"])
-        rescue
-          source_version = self.class.split(str).first || "0"
-          if source_version == "0" # it probably wasn't encrypted
-            return str
-          elsif source_version == "2" # tried with an alt key, see if regular v2 key works
-            decrypt(str)
-          else
-            raise
-          end
-        end
+    def self.recrypt(str, prior_key = nil)
```

Comment:
so the prior key used to be loaded out of setting an alt key.

you're changing this interface.

It is only changed in the one spot for fix_auth, but would be nice to not make breaking changes across repos.

Reply:
yes, I changed this interface because with the removal of legacy keys we don't really need to keep a complicated Hash of keys at the class level.  Since the alt key was only used for recrypted I made it a parameters.  This is a breaking change PR for a lot of other reasons, so I'm ok with changing it.  (Note, I have more PRs coming for other things like vault support, which will break it further, so I figured we might as well pick a good interface now)